### PR TITLE
Fixes to the vector-space build process

### PR DESCRIPTION
**Pull request by rspeer**
While my hopes of making this build in a nearly-reasonable 24 GB of RAM must be set aside, I at least made it work in 30 GB. Various array operations are now done in-place instead of making copies of large arrays.

Also fixed some things that were making the vectors not come out as good as ones we've previously released. Language-balancing is gone for now, and the sampling strategy in `merge.py` is back to what it was.

### CODE REVIEW

File: conceptnet5/vectors/debias.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 from conceptnet5.vectors import standardized_uri, normalize_vec
 from conceptnet5.vectors.transforms import l2_normalize_rows
+import pandas as pd
 import numpy as np
 from sklearn import svm
+from sklearn.preprocessing import normalize
```
Comment (jlowryduda) [inline conceptnet5/vectors/debias.py]: Let's organize these imports according to PEP8 guidelines.

File: conceptnet5/vectors/debias.py [hunk 2]
```diff
@@ -390,14 +392,17 @@ def reject_subspace(frame, vecs):
...
     for vec in vecs:
         if not np.isnan(vec).any():
             vec = normalize_vec(vec)
             projection = current_array.dot(vec)
-            current_array -= np.outer(projection, vec)
+            np.subtract(current_array, np.outer(projection, vec), out=current_array)
 
-    return l2_normalize_rows(current_array, offset=1e-9)
+    current_array = pd.DataFrame(current_array, index=frame.index)
+    normalize(current_array.values, norm='l2', copy=False)
```
Comment (jlowryduda) [inline conceptnet5/vectors/debias.py]: I'm not quite sure why, but when I ran it, `normalize(current_array.values, norm='l2', copy=False)` didn't actually modify `current_array`. Interestingly, it seems like `normalize(result.values, norm='l2', copy=False)` in line 498 actually modifies `result`.

File: conceptnet5/vectors/merge.py [hunk 3]
```diff
@@ -17,57 +15,119 @@ def dataframe_svd_projection(frame, k):
...
+
+    # Find a subset of the labels that we'll use for calculating the
+    # dimensionality-reduced version. The labels we particularly care about
+    # are single words in our CORE_LANGUAGES. Even those are too numerous,
+    # so we take an arbitrary 1/n sample of them, where n is given by the
+    # `subsample` parameter.
+    filtered_labels = pd.Series([
+        label for (i, label) in enumerate(joined.index)
+        if i % subsample == 0 and '_' not in label
+        and label.split('/')[2] in CORE_LANGUAGES
```
Comment (jlowryduda) [inline conceptnet5/vectors/merge.py]: I think `get_language` can be used here.

File: conceptnet5/vectors/debias.py (hunk 2 above)
Comment (rspeer) [inline conceptnet5/vectors/debias.py]: So wait, the same code in two different circumstances either does or does not modify the array?

Comment (jlowryduda) [inline conceptnet5/vectors/debias.py]: That's what I'm seeing, but that sounds like a thing that shouldn't be happening. Can you confirm that `current_array` is modified by `normalize`?

Comment (rspeer) [inline conceptnet5/vectors/debias.py]: It works in the simple examples I'm trying. Now I'm worried that whether the inner matrix gets normalized depends on something complicated involving memory aliasing.

I notice that this particular example could be simplified. I could be normalizing before converting `current_array` into a DataFrame.
